ledger/participant-integration-api/src/main/scala/platform/LedgerApiServer.scala

```diff
@@ -50,7 +51,9 @@ class LedgerApiServer(
     //          Currently, we provide this flag outside the HOCON configuration objects
     //          in order to ensure that participants cannot be configured to accept explicitly disclosed contracts.
     explicitDisclosureUnsafeEnabled: Boolean = false,
-    rateLimitingInterceptor: Option[RateLimitingInterceptor] = None,
+    rateLimitingInterceptor: Option[
+      QueueAwareExecutionContextExecutorService => RateLimitingInterceptor
```

Comment:
Seems strange, the executor service is created outside the LedgerApiServer.  Seems cleaner to make the association outside of the Ledger Api Server (as now).

Reply:
The problem is that the executor service is actually created in the `Indexer`, which is created as part of the `LedgerApiServer`. This is why we need to pass a way to create an interceptor from an existing executor service.
To refactor this we would need to refactor the start-up phase or move the executor service creation for the indexer a few levels up. I found this change to be the less disruptive from all of them.

Reply:
Yes.